### ISSUE TITLE
Incorrect results dirs

### ISSUE THREAD

**Issue by pkking** [fedora-copr/copr#2567]
Hi team, 
I notice a strange error:
- when i submit a build though cli with: `copr build-package  --name python-abimap test`，the builder log will live in `{buildid}-{package_name}/builder-live.log.tar.gz` dir
![image](https://user-images.githubusercontent.com/2853724/223357265-d0a7f8d1-37d9-4ecd-abee-da0b9fd03af9.png)
 but the log `url` in frontend page is just `{buildid}/builder-live.log.tar.gz` 
![image](https://user-images.githubusercontent.com/2853724/223357526-14ec7c8a-266a-4a70-8cc2-d406925dc9c9.png)
- when i rebuild directly through rebuild page, it works as expected
- when i tried the same command using the offical copr instance, it works as expected

It can be reproduced in latest `docker-compose` setup and have no idea why, any help appreciated!

**Comment by praiskup** [comment]
Thank you for the question.

Ok, so the problem is that backend stores the data to `srpm-builds/<build_id>-`**`<package_name>`**, instead of just `srpm-builds/<build_id>`.  [This code](https://github.com/fedora-copr/copr/blob/8fe013ebf4066ca31b2e919b809229800d334fa6/backend/copr_backend/helpers.py#L691-L695) does it.

It seems that `package_name` is for some reason defined by fronend in the srpm task, but [it shouldn't](https://github.com/fedora-copr/copr/blob/8fe013ebf4066ca31b2e919b809229800d334fa6/frontend/coprs_frontend/coprs/views/backend_ns/backend_general.py#L193-L236).

Shrugs.  Any idea?

**Comment by pkking** [comment]
Seem this [code](https://github.com/fedora-copr/copr/blob/fceb79456e7f7e9a41562f58d0903a262566b505/frontend/coprs_frontend/coprs/views/backend_ns/backend_general.py#L228) break the rule, now srpm build task will always return a valid package name, which cause the problem

**Comment by pkking** [comment]
Anyway, maybe the [code](https://github.com/fedora-copr/copr/blob/fceb79456e7f7e9a41562f58d0903a262566b505/frontend/coprs_frontend/coprs/models.py#L1134-L1138) should flow the TODO hint, do not hardcode path and honor the `build.result_dir` all the time.

**Comment by praiskup** [comment]
Good point, thank you for spotting this before the actual release.  Yes, we should assure that
we set the result_dir appropriately.
